ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/WebSocketTest.kt

```diff
@@ -142,6 +146,29 @@ class WebSocketTest : ClientLoader(100000) {
         }
     }
 
+    @Serializer(forClass = Data::class)
+    object DataSerializer
+
+    @Test
+    fun testWebSocketSerializationWithCustomSerializer() = clientTests(listOf("Android", "Apache", "Curl")) {
+        config {
+            WebSockets {
+                contentConverter = KotlinxWebsocketSerializationConverter(Json)
+            }
+        }
+
+        test { client ->
+            client.webSocket("$TEST_WEBSOCKET_SERVER/websockets/echo") {
+                repeat(TEST_SIZE) {
+                    val originalData = Data("hello")
+                    sendSerialized(originalData, DataSerializer)
```

Comment:
what is the use case of passing serializer here? can't it be registered in module or through `@UseSerializers` annotation?

Reply:
I will add it in another PR and focus this PR on Websockets only.

Reply:
BTW: have you tried to register external serializer for 3rd party class via `SerializersModule` via `contextual` call?. According to code it should just work by fetching serialiaer by `KClass`
Or can you at least provide an example of what you want to archive, and it fails

Reply:
I want to archive this:
```
private val json = Json {
    SerializersModule {
        contextual(Data::class, DataSerializer)
    }
}

fun main() {
    val main = Data(42)
    val mainJson = json.encodeToString(main)
    println(mainJson)
}

data class Data(val value: Int)

@Serializer(forClass = Data::class)
object DataSerializer
```
And adding this to contextual does not work.

Reply:
I deleted my old reproducer and at the moment reproducing the case does not work. But adding all possibilities via `contextual` or `polymorphic` to the serializer module is annoying, so passing the serializer directly at usage is nicer IMHO. And the implementation is quite forward for web sockets.

Reply:
I think that both ways are little annoying :)
But, even it's quite forward for websockets, looks reasonable to add the same to http endpoints, but there it will be not such easy.

And the main my concern about, why it's not good - calling this method will fail, if there is non kx.serialization converter.
Or f.e. if there will be some custom converter, which delegates to kotlinx.serialization converter, but uses another encoding for some types. So using `serializersModule` will work in both cases, but providing explicit serializer will work only in case of plain kx.serialization converter - such an inconsistency isn't so good in my opionion.

And in this specific use case, may be it will be much easier to not use this plugin at all, and do serialization handling manually, or write a simple plugin for your use case? :)